### PR TITLE
Gawk input suffix

### PR DESCRIPTION
**Pull request by muffato**
1. Transition away from `input.suffix` into a proper input `val`, still falling back to the extension of the first file
   This is a *breaking change* as the shape of the input channels changes.
2. Support empty input lists (when suffix is not provided)

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Broadcast software version numbers to `topic: versions` - [See version_topics](https://nf-co.re/blog/2025/version_topics)
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/gawk/main.nf [hunk 1]
```diff
@@ -23,7 +23,14 @@ process GAWK {
     def args  = task.ext.args  ?: '' // args is used for the main arguments of the tool
     def args2 = task.ext.args2 ?: '' // args2 is used to specify a program when no program file has been given
     prefix    = task.ext.prefix ?: "${meta.id}"
-    suffix    = task.ext.suffix ?: "${input.collect{ file -> file.getExtension()}.get(0)}" // use the first extension of the input files
+
+    if (!suffix) {
```
Comment (erikrikarddaniel) [inline modules/nf-core/gawk/main.nf:27]: Pipelines that update this module will still have `ext.suffix` in their `modules.config`. If they pass `""` or `[]` as the new input, `ext.suffix` is now ignored without a warning, and the output extension changes to the input's. Would `task.ext.suffix` as a middle fallback be worth keeping for a while? For example, `suffix = suffix ?: task.ext.suffix ?: input[0].extension`. Or at least a line in the PR description for pipeline developers. Not blocking.

File: modules/nf-core/gawk/meta.yml [hunk 2]
```diff
@@ -53,8 +56,7 @@ output:
           type: file
           description: The output file - if using shell redirection, specify the
             name of this file using `ext.prefix` and the extension using
-            `ext.suffix`. Otherwise, ensure the awk program produces files with
-            the extension in `ext.suffix`.
+            `suffix` (falling back to the extension of the first input file.
```
Comment (erikrikarddaniel) [inline modules/nf-core/gawk/meta.yml]: Unclosed parenthesis.

```suggestion
            `suffix` (falling back to the extension of the first input file).
```

File: modules/nf-core/gawk/tests/main.nf.test [hunk 3]
```diff
@@ -208,4 +207,52 @@ nextflow_process {
             assert process.errorReport.contains("Input and output names are the same, set prefix in module configuration to disambiguate!")
         }
     }
+
+    test("No input file but suffix provided") {
```
Comment (erikrikarddaniel) [inline modules/nf-core/gawk/tests/main.nf.test:235]: No test covers the fallback that existing callers rely on: `suffix` empty, input present. I ran this locally with docker and it passes, with output `test.fai`:

```groovy
    test("Suffix falls back to the input extension") {
        when {
            params {
                gawk_args2  = '\'BEGIN { FS = OFS = "\t"}; { print \$1, "0", \$2 }\''
            }
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true),
                    "",
                ]
                input[1] = []
                input[2] = false
                """
            }
        }

        then {
            assert process.success
            assert snapshot(sanitizeOutput(process.out)).match()
        }
    }
```

**Review by erikrikarddaniel — APPROVED** [review]
Reviewed by Claude: 

I checked that every GAWK caller in the repo was updated: all 17 files that include the module are in this PR. Each passes the suffix its ext.suffix used to set. No ext.suffix for GAWK is left in any config.

I also checked the one path the tests don't cover: an empty suffix with an input file. It works, because the reassigned suffix reaches the output glob and the output is test.fai. A small test for that is suggested inline, plus a migration question and a typo.

File: modules/nf-core/gawk/tests/main.nf.test (hunk 3 above)
Comment (muffato) [reply modules/nf-core/gawk/tests/main.nf.test:235]: Thanks for the contribution ! I've added your test as-is

File: modules/nf-core/gawk/main.nf (hunk 1 above)
Comment (muffato) [reply modules/nf-core/gawk/main.nf:27]: I'm not keen on that. My view is that the breaking input change means that people will notice.

Other modules transitioned away from `ext.suffix` recently don't have a grace period either.
